src/ViewModels/src/CloudExplorer/CloudExplorerViewModel.cs

```diff
@@ -15,6 +17,7 @@ public class CloudExplorerViewModel : AbstractViewModel, ICloudExplorerViewModel
         internal static readonly string _stopAppErrorMsg = "Encountered an error while stopping app";
         internal static readonly string _startAppErrorMsg = "Encountered an error while starting app";
         internal static readonly string _deleteAppErrorMsg = "Encountered an error while deleting app";
+        private static IErrorDialog _dialogService;
```

Comment:
You've scoped this field as `static` so there's only ever one instance, yet you initialize in constructor.  So if you construct 2 or more of these objects, each will overwrite the previous service instance.